File: serverless-aws/node/custom_tag_and_context/custom_tag_and_context.js
/**
 *******************************************************************************
 * File name   : custom_tag_and_context.js
 * Description : This file contains code that instruments Sentry custom Tags and
 *               Context.
 *******************************************************************************
 **/

// Import the Sentry module.
const Sentry = require("@sentry/node");
// Configure the Sentry SDK.
Sentry.init({
  dsn: "<Your dsn>",
});

let env_value = process.env.ENV_VAR; // Fetching value of Environment variable set in Lambda function.

Sentry.configureScope(function (scope) {
  scope.setLevel("fatal");
  scope.setTag("custom_tag", "custom_tag_value");
  scope.setExtra("extra_context", "extra_context_value");
  scope.setExtra("ENV_VAR", env_value);
});

// below is the faulty code, aFunctionThatMightFail() function is not exist.
try {
  aFunctionThatMightFail(); // Call undefined function.
} catch (e) {
  Sentry.captureException(e); // Capture the exception in the Sentry dashboard.
  Sentry.flush(2000);
}

exports.handler = function (event, context, callback) {
  const response = {
    statusCode: 200,
    body: "Custom tags and events",
  };

  callback(null, response);
};

File: aws-lambda/python/custom_tag_and_context/custom_tag_and_context.py
# /**
# *******************************************************************************
# * File name   : custom_tag_and_context.py
# * Description : This file contains code that instruments handled exception and
#                 creates custom tag & context
# *******************************************************************************
# **/

# Import Sentry library
import sentry_sdk
from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration
from sentry_sdk import configure_scope

import os

# Configure Sentry SDK
sentry_sdk.init(
    dsn="<your DSN>",
    integrations=[AwsLambdaIntegration()]
)

# Fetching value of Environment variable set in Lambda function
env_var_value = os.environ['ENV_VAR']


def lambda_handler(event, context):
    """Lambda function which raises an exception and creates Custom tag &
    context.

    Args:
        event (dict): Parameter to pass in event data to the handler.
        context (bootstrap.LambdaContext): Parameter to provide runtime information to the handler.
    """
    with configure_scope() as scope:
        scope.set_tag("custom_tag", "Test tag value")
        scope.set_extra("ENV_VAR", env_var_value)
    try:
        division_by_zero = 1 / 0
    except Exception as e:
        # handle ZeroDivisionError exception
        print(e)
        raise
